src/layout/core/graph_tree.rs

```diff
@@ -684,6 +715,33 @@ impl InnerTree {
             node_ix = parent_ix;
         }
     }
+
+    /// Gets the next sibling to focus on, assuming child_ix would be removed from parent_ix.
+    /// If there are no other siblings to focus on, parent_ix is returned.
+    ///
+    /// # Panics
+    /// This will panic if `child_ix` is not a child of `parent_ix`
+    pub fn next_sibling(&self, parent_ix: NodeIndex, child_ix: NodeIndex) -> NodeIndex {
+        let children = self.children_of(parent_ix);
+        let mut prev_index = None;
+        for (index, sibling_ix) in children.iter().enumerate() {
+            if child_ix == *sibling_ix {
+                prev_index = Some(index);
```

Comment:
Should break here.

In fact this could probably be rewritten to be more functional in general.